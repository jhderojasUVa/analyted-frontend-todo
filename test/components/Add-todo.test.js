import { describe, it, expect } from 'vitest';
import { mount } from '@vue/test-utils';

import AddTodo from '../../src/components/add-todo/Add-todo.vue';

describe('addtodo component', () => {
    it('renders', () => {
        const description = 'foo';

        const wrapper = mount(AddTodo, { props: { description } });

        expect(wrapper).toMatchSnapshot();
    });
    describe('methods', () => {
        it('addTodo emits', () => {
            const description = 'foo';
            const wrapper = mount(AddTodo, { props: { description: 'something' } });
    
            wrapper.vm.$emit('addTodo', description);
    
            expect(wrapper.emitted().addTodo).toStrictEqual([[description]]);
        });
    })
    it('add a new works', async () => {
        const description = 'foo';

        const wrapper = mount(AddTodo, { props: { description: '' } });

        // search for input and button
        const input = wrapper.find('input');
        const button = wrapper.find('button');

        // set new value
        await input.setValue(description);

        button.trigger('click');

        wrapper.vm.$nextTick();

        // check
        expect(wrapper.find('input').element.value).toBe(description);
        expect(wrapper.vm.description).toBe(description);
        expect(wrapper.emitted().click).not.toBe(undefined);
    });
});
