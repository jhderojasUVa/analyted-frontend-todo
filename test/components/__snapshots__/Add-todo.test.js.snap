// Vitest Snapshot v1

exports[`addtodo component > renders 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        data-v-eb1153dd=""
        description="foo"
      >
        <input
          data-v-eb1153dd=""
          maxlength="50"
          placeholder="Add your new todo here"
          required=""
          size="20"
          type="text"
        />
        <button
          data-v-eb1153dd=""
        >
          Add
        </button>
      </div>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {},
      "da": null,
      "data": {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {},
      "refs": {
        "VTU_COMPONENT": {
          "add": [Function],
          "description": "",
          "hasOwnProperty": [Function],
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": [],
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": {},
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "add": 4,
            "constructor": 0,
            "description": 2,
            "hasOwnProperty": 4,
            "toJSON": 0,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {
            "description": "foo",
          },
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {
            "add": [Function],
            "description": "",
            "hasOwnProperty": [Function],
          },
          "da": null,
          "data": {
            "description": "",
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": {},
          "propsDefaults": {},
          "propsOptions": [
            {},
            [],
          ],
          "provides": {},
          "proxy": {
            "add": [Function],
            "description": "",
            "hasOwnProperty": [Function],
          },
          "refs": {},
          "render": [Function],
          "renderCache": [
            [Function],
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": [],
            "effects": [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": {},
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": [
                  {
                    "arg": undefined,
                    "dir": {
                      "beforeUpdate": [Function],
                      "created": [Function],
                      "mounted": [Function],
                    },
                    "instance": {
                      "add": [Function],
                      "description": "",
                      "hasOwnProperty": [Function],
                    },
                    "modifiers": {},
                    "oldValue": undefined,
                    "value": "",
                  },
                ],
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <input
                  data-v-eb1153dd=""
                  maxlength="50"
                  placeholder="Add your new todo here"
                  required=""
                  size="20"
                  type="text"
                />,
                "key": null,
                "patchFlag": 512,
                "props": {
                  "maxlength": "50",
                  "onUpdate:modelValue": [Function],
                  "placeholder": "Add your new todo here",
                  "required": "",
                  "size": "20",
                  "type": "text",
                },
                "ref": null,
                "scopeId": "data-v-eb1153dd",
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Add",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <button
                  data-v-eb1153dd=""
                >
                  Add
                </button>,
                "key": null,
                "patchFlag": 0,
                "props": {
                  "onClick": [Function],
                },
                "ref": null,
                "scopeId": "data-v-eb1153dd",
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "button",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": [
                  {
                    "arg": undefined,
                    "dir": {
                      "beforeUpdate": [Function],
                      "created": [Function],
                      "mounted": [Function],
                    },
                    "instance": {
                      "add": [Function],
                      "description": "",
                      "hasOwnProperty": [Function],
                    },
                    "modifiers": {},
                    "oldValue": undefined,
                    "value": "",
                  },
                ],
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <input
                  data-v-eb1153dd=""
                  maxlength="50"
                  placeholder="Add your new todo here"
                  required=""
                  size="20"
                  type="text"
                />,
                "key": null,
                "patchFlag": 512,
                "props": {
                  "maxlength": "50",
                  "onUpdate:modelValue": [Function],
                  "placeholder": "Add your new todo here",
                  "required": "",
                  "size": "20",
                  "type": "text",
                },
                "ref": null,
                "scopeId": "data-v-eb1153dd",
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
            ],
            "dynamicProps": null,
            "el": <div
              data-v-eb1153dd=""
              description="foo"
            >
              <input
                data-v-eb1153dd=""
                maxlength="50"
                placeholder="Add your new todo here"
                required=""
                size="20"
                type="text"
              />
              <button
                data-v-eb1153dd=""
              >
                Add
              </button>
            </div>,
            "key": null,
            "patchFlag": 16,
            "props": {
              "description": "foo",
            },
            "ref": null,
            "scopeId": "data-v-eb1153dd",
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/add-todo/Add-todo.vue",
            "__scopeId": "data-v-eb1153dd",
            "components": {},
            "data": [Function],
            "methods": {
              "add": [Function],
            },
            "name": "AddTodo",
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          data-v-eb1153dd=""
          description="foo"
        >
          <input
            data-v-eb1153dd=""
            maxlength="50"
            placeholder="Add your new todo here"
            required=""
            size="20"
            type="text"
          />
          <button
            data-v-eb1153dd=""
          >
            Add
          </button>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "description": "foo",
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "/mnt/c/Projects/analyted/analyted-frontend-todo/src/components/add-todo/Add-todo.vue",
          "__scopeId": "data-v-eb1153dd",
          "components": {},
          "data": [Function],
          "methods": {
            "add": [Function],
          },
          "name": "AddTodo",
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          data-v-eb1153dd=""
          description="foo"
        >
          <input
            data-v-eb1153dd=""
            maxlength="50"
            placeholder="Add your new todo here"
            required=""
            size="20"
            type="text"
          />
          <button
            data-v-eb1153dd=""
          >
            Add
          </button>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.37",
  },
  "__setProps": [Function],
  "componentVM": {
    "add": [Function],
    "description": "",
    "hasOwnProperty": [Function],
  },
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <div
    data-v-eb1153dd=""
    description="foo"
  >
    <input
      data-v-eb1153dd=""
      maxlength="50"
      placeholder="Add your new todo here"
      required=""
      size="20"
      type="text"
    />
    <button
      data-v-eb1153dd=""
    >
      Add
    </button>
  </div>,
}
`;
